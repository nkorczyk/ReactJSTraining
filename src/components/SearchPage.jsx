import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header';
import Results from './Results';
import Content from './Content';
import Footer from './Footer';
import ErrorBoundary from './ErrorBoundary';

const SearchPage = ({ movies, searchby, sortby }) => {
  return (
    <div>
      <Header searchby={searchby} />
      <Results items={movies.length} sortby={sortby} />
      <ErrorBoundary>
        <Content movies={movies} />
      </ErrorBoundary>
      <Footer />
    </div>
  )
}

Content.propTypes = {
  movies: PropTypes.array,
  searchby: PropTypes.func,
  sortby: PropTypes.func,
};

export default SearchPage;
