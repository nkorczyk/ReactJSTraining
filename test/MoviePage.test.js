import React from 'react';
import { mount } from 'enzyme';
import SearchPage from "../src/components/SearchPage";

describe('should render SearchPage component', () => {
  const movies_data = [{
    "id": 321612,
    "title": "Beauty and the Beast",
    "tagline": "Be our guest.",
    "vote_average": 6.8,
    "vote_count": 7861,
    "release_date": "2017-03-16",
    "poster_path": "https://image.tmdb.org/t/p/w500/tWqifoYuwLETmmasnGHO7xBjEtt.jpg",
    "overview": "A live-action adaptation of Disney's version",
    "genres": [
      "Family",
      "Fantasy",
      "Romance"
    ],
  }];
  const element = mount(<SearchPage movies={movies_data} />);
  test('Snapshot test with default props', () => {
    expect(element).toMatchSnapshot();
  });
});
