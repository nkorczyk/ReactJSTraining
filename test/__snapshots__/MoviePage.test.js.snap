// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchPage component Snapshot test with default props 1`] = `undefined`;
